woocommerce/woocommerce-admin
# Inbox notification: Add dismiss functionality

Fixes #partially-4099

This PR adds the dismiss functionality to the inbox notifications.
A soft delete for inbox notifications is implemented here. The client will be able to delete one notification or all of them. A Snackbar will appear after the soft delete bringing the possibility to "Undo" action.

_Please be sure the [PR 4258](https://github.com/woocommerce/woocommerce-admin/pull/4258) has been approved before approving this one_

> **_NOTE:_**  As this is a part of a big functionality this branch will be merged with the branch `add/4099`, and it will be that branch that will be merged with `master`.
To make the code reviewing easier, this PR only has the code referring to the functionality described in the title.
If you want to test everything together, all the changes are merged in the branch: `add/4099_test_branch`.


### Screenshots
![Screen Capture on 2020-05-15 at 13-37-13](https://user-images.githubusercontent.com/1314156/82076342-14796080-96b4-11ea-9516-df790f702cb9.gif)

### Detailed test instructions:

- First, it's necessary to let dbDelta do its job updating the database (adding the `is_deleted`, `layout` and `image` columns and removing the `icon` column from the table `wp_wc_admin_notes`).
- To be able to see the inbox notifications, press Inbox in the navigation bar.
- The list of inbox notifications should be visible.
- Press `Dismiss` in any message.
- Choose `Dismiss this message`.
- A confirmation modal will be visible. Press `Yes, I'm sure`.
- The message shouldn't be visible anymore.
- Do the same choosing the option `Dismiss all messages`



### Changelog Note:
Dev: Added dismiss functionality to inbox notifications
